src/chai/ManagedArray.inl

```diff
@@ -393,16 +395,24 @@ void ManagedArray<T>::move(ExecutionSpace space) const
      m_active_pointer = m_active_base_pointer + m_offset;
 
      CHAI_LOG(Debug, "Moved to " << m_active_pointer);
+#if defined(CHAI_ENABLE_UM)
+    if (m_pointer_record->m_last_space == UM) {
+    } else
+#endif
+#if defined(CHAI_ENABLE_PINNED)
+    if (m_pointer_record->m_last_space == PINNED) {
+    } else 
+#endif
      if (!std::is_const<T>::value) {
        CHAI_LOG(Debug, "T is non-const, registering touch of pointer" << m_active_pointer);
        m_resource_manager->registerTouch(m_pointer_record);
```

Comment:
Doesn't pinned sort of require a touch? If we write on the device, don't we want to synchronize before it is used on the host again?

Reply:
for pinned memory, we only need to track whether globally we need to synchronize. This branch takes the approach of the ArrayManager marking pinned memory as potentially needing a sync whenever a GPU space is being used, and the sync will only occur if a non-const pinned pointer is needed on the host. It's not quite an optimal case, e.g. - non -const pin memory used on GPU, host context, gpu context without use of that pointer, another host context with that pointer would then trigger another synchronize. That said, it's pretty close to optimal - it does not trigger a synchronize for every pointer, for instance. In practice this is working very well in our application.